crates/store/re_dataframe2/src/query.rs

```diff
@@ -726,3 +739,681 @@ impl<'a> QueryHandle<'a> {
+            let expected = unindent::unindent(
+                "\
+                [
+                    Int64[None, 1, 2, 3, 4, 5, 6, 7],
+                    Int64[None, 1, 2, 3, 4, 5, 6, 7],
+                    NullArray(8),
+                ]\
+                ",
+            );
+
+            similar_asserts::assert_eq!(expected, got);
+        }
+
+        // only components (+ duplication)
+        {
+            let mut query = QueryExpression2::new(timeline);
+            query.selection = Some(vec![
+                ColumnSelector::Component(ComponentColumnSelector {
+                    entity_path: entity_path.clone(),
+                    component: MyColor::name(),
+                    join_encoding: Default::default(),
+                }),
+                ColumnSelector::Component(ComponentColumnSelector {
+                    entity_path: entity_path.clone(),
+                    component: MyColor::name(),
+                    join_encoding: Default::default(),
+                }),
+                ColumnSelector::Component(ComponentColumnSelector {
+                    entity_path: "non_existing_entity".into(),
+                    component: MyColor::name(),
+                    join_encoding: Default::default(),
+                }),
+                ColumnSelector::Component(ComponentColumnSelector {
+                    entity_path: entity_path.clone(),
+                    component: "AComponentColumnThatDoesntExist".into(),
+                    join_encoding: Default::default(),
+                }),
+            ]);
+            eprintln!("{query:#?}:");
+
+            let query_handle = query_engine.query(query.clone());
+            let dataframe = concatenate_record_batches(
+                query_handle.schema().clone(),
+                &query_handle.into_batch_iter().collect_vec(),
+            );
+            eprintln!("{dataframe}");
+
+            let got = format!("{:#?}", dataframe.data.iter().collect_vec());
+            let expected = unindent::unindent(
+                "\
+                [
+                    ListArray[None, None, None, [2], [3], [4], None, [6]],
+                    ListArray[None, None, None, [2], [3], [4], None, [6]],
+                    NullArray(8),
+                    NullArray(8),
+                ]\
+                ",
+            );
+
+            similar_asserts::assert_eq!(expected, got);
+        }
+
+        Ok(())
+    }
+
+    #[test]
+    fn view_contents_and_selection() -> anyhow::Result<()> {
+        re_log::setup_logging();
+
+        let store = create_nasty_store()?;
+        eprintln!("{store}");
+        let query_cache = QueryCache::new(&store);
+        let query_engine = QueryEngine {
+            store: &store,
+            cache: &query_cache,
+        };
+
+        let entity_path: EntityPath = "this/that".into();
+        let timeline = Timeline::new_sequence("frame_nr");
+
+        // only components (+ duplication)
```

Comment:
I'm not seeing the duplication here.

Reply:
aaaah i didnt mean to check duplication _again_ here, it's just the comment that's wrong :relieved: